rules/repo.bzl

```diff
@@ -4,12 +4,17 @@
 
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-def http_archive_or_local(**kwargs):
-    local = kwargs.pop("local", None)
-    if local:
-        native.local_repository(
+def http_archive_or_local(local = None, build_file = None, **kwargs):
+    if not local:
+        http_archive(build_file = build_file, **kwargs)
+    elif build_file:
+        native.new_local_repository(
             name = kwargs.get("name"),
             path = local,
+            build_file = build_file,
         )
     else:
-        http_archive(**kwargs)
```

Comment:
I assume http_archive doesn't need or properly handle kwargs here so it's intentionally omitted.

Reply:
This is an accidental duplicate comment, right?

Reply:
Typo, sorry, same question about the local_repository rule (but then I apparently looked at http_archive and typed that instead)